var/spack/repos/builtin/packages/oce/package.py

```diff
@@ -44,6 +44,8 @@ class Oce(Package):
 
     variant('tbb', default=True,
             description='Build with Intel Threading Building Blocks')
+    variant('X11', default=False,
+            description='Build with X11 enabled')
```

Comment:
For the record, there is a Mesa package in Spack. The last time I tried to build it, it crashed, so it isn't very stable. But there are at least 3 open PRs that are working on fixing that.